bugzooka/integrations/jira_client.py

```diff
@@ -0,0 +1,293 @@
+                        ),
+                        "created": fields.get("created", ""),
+                        "updated": fields.get("updated", ""),
+                    }
+                )
+
+            return issues
+
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Error searching Jira issues: {e}")
+            if hasattr(e, "response") and e.response is not None:
+                logger.error(f"Response status: {e.response.status_code}")
+                logger.error(f"Response content: {e.response.text}")
+            raise Exception(f"Failed to search Jira issues: {str(e)}")
+
+    def get_issue(self, issue_key: str) -> Optional[dict]:
+        """Get a specific issue by key."""
+        try:
+            response = requests.get(
+                f"{self.base_url}/rest/api/2/issue/{issue_key}",
+                headers=self.headers,
+                timeout=30,
+            )
+            response.raise_for_status()
+
+            data = response.json()
+            fields = data.get("fields", {})
+
+            return {
+                "key": data.get("key"),
+                "summary": fields.get("summary", ""),
+                "description": fields.get("description", ""),
+                "status": (fields.get("status") or {}).get("name", ""),
+                "assignee": (fields.get("assignee") or {}).get("displayName", ""),
+                "reporter": (fields.get("reporter") or {}).get("displayName", ""),
+                "created": fields.get("created", ""),
+                "updated": fields.get("updated", ""),
+            }
+
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Error getting Jira issue {issue_key}: {e}")
+            return None
+
+
+def get_allowed_projects() -> List[str]:
+    """Get allowed Jira projects from environment variable.
+    Returns:
+        List of allowed project keys
+    Raises:
+        ValueError: If no valid projects are configured
+    """
+    projects_env = os.environ.get("JIRA_ALLOWED_PROJECTS")
+    if not projects_env:
+        raise ValueError("JIRA_ALLOWED_PROJECTS environment variable is not set")
+
+    allowed = [p.strip() for p in projects_env.split(",") if p.strip()]
+
+    logger.debug(f"Allowed Jira projects: {allowed}")
+
+    if not allowed:
+        raise ValueError("JIRA_ALLOWED_PROJECTS must contain at least one project")
+
+    return allowed
+
+
+def get_jira_client() -> JiraClient:
+    """Get Jira client from environment variables."""
+    base_url = os.environ.get("JIRA_BASE_URL")
+    token = os.environ.get("JIRA_TOKEN")
+
+    logger.debug(f"JIRA_BASE_URL: {base_url}")
+    logger.debug(f"JIRA_TOKEN: {token[:10] if token else 'None'}...")
+
+    if not base_url or not token:
+        raise ValueError(
+            "Missing Jira credentials: JIRA_BASE_URL and JIRA_TOKEN must be set in environment"
+        )
+
+    return JiraClient(base_url, token)
+
+
+@mcp.tool()
```

Comment:
This is jeopardising the whole purpose of bring you own MCP model. Tools should be defined somewhere else and only client calls should happen from here